src/cloudflare/internal/test/d1/d1-api-test.wd-test

```diff
@@ -9,7 +9,7 @@ const unitTests :Workerd.Config = (
           (name = "worker", esModule = embed "d1-api-test.js")
         ],
         compatibilityDate = "2023-01-15",
-        compatibilityFlags = ["nodejs_compat"],
+        compatibilityFlags = ["nodejs_compat", "enable_d1_with_sessions_api"],
```

Comment:
It took me some time to figure out the Workerd configuration, but it seems simpler now actually too. I added the new tests into a new test suite and extracted the common tests into a shared file.

I left the existing file without the sessions API tests because it's used by the Eyeball E2E tests by its name. I will adjust those too later to include Sessions tests.